docs/pages/machine-id/guides/github-actions-kubernetes.mdx

```diff
@@ -0,0 +1,221 @@
+---
+title: Using Machine ID With GitHub Actions and Kubernetes
+description: A tutorial for using Machine ID with GitHub Actions and Kubernetes
+---
+
+<Details
+  title="Version warning"
+  opened={true}
+  scope={["oss", "enterprise"]}
+  scopeOnly={true}
+  min="11.0"
+>
+  Machine ID for GitHub Actions is available starting from Teleport `v11.0`.
+</Details>
+
+GitHub Actions is a popular CI/CD platform that works as a part of the larger
+GitHub ecosystem.
+
+In this guide, we will use Teleport Machine ID and Kubernetes Access to allow a
+GitHub Actions workflow to securely connect to a Kubernetes cluster without the
+need for long-lived secrets.
+
+Teleport supports secure joining on both GitHub-hosted and self-hosted GitHub
+Actions runners as well as GitHub Enterprise Server.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A Kubernetes cluster connected to your Teleport cluster. If you do not already
+have one configured, try our
+[Kubernetes Access getting started guide.](../../kubernetes-access/getting-started.mdx)
+In our examples, this Kubernetes cluster will be named `my-kubernetes-cluster`.
+- A local workstation with `tsh` and `tctl` authorized against your Teleport
+cluster as a user with the privileges to create configuration resources.
+- A GitHub repository with GitHub Actions enabled. This guide uses the example
+`gravitational/example`, and this value should be replaced with your own unique
+repository.
+
+### 1/4. Create a join token for GitHub Actions
+
+(!docs/pages/includes/machine-id/github-actions-create-token.mdx!)
+
+## Step 2/4. Create a Machine ID bot
+
+With the join token for the GitHub Actions workflow created, we now need to
+create a Machine ID bot that the token will grant access to. A Machine ID bot is
+a special type of Teleport user designed for access by machines, and can
+authenticate using a join token rather than forms of authentication more
+suitable to users (e.g such as SSO.)
+
+Before creating the bot, first you must create a role in Teleport which grants
+the bot access to your Kubernetes clusters and defines the Kubernetes RBAC
+groups that will be added to your bot's requests to the Kubernetes cluster.
+
+Create `role.yaml` and insert the following contents:
+
+```yaml
+kind: role
+metadata:
+  name: github-demo-kube-access
+version: v5
+spec:
+  allow:
+    kubernetes_labels:
+      # This grants access to any Kubernetes cluster attached to your Teleport
+      # cluster.
+      '*': '*'
+    kubernetes_groups:
+      # This states that the `github-demo` group will be applied to your bot's
+      # requests to your Kubernetes cluster. In a later step, we will bind this
+      # group to a Kubernetes role with a ClusterRoleBinding.
+      - github-demo
+  deny: {}
+```
+
+Apply this to your Teleport cluster:
```

Comment:
You could be a little more specific with "Apply this file", or more direct with something like "create this object/role".